components/multi-platform-controller/base/host-config-chart/templates/host-config.yaml

```
@@ -972,6 +972,27 @@ data:
   dynamic.linux-extra-fast-amd64.allocation-timeout: "1200"
   {{ end }}
 
+  {{- if hasKey .Values.dynamicConfigs "macos-mac2metal-arm64" }}
+  {{- $config := index .Values.dynamicConfigs "macos-mac2metal-arm64" | default (dict) }}
+  dynamic.macos-mac2metal-arm64.type: {{ index $config "type" | default "aws" | quote }}
+  dynamic.macos-mac2metal-arm64.region: {{ index $config "region" | default "us-east-1" | quote }}
+  dynamic.macos-mac2metal-arm64.ami: {{ default (index $amd "ami") $config.ami | quote }}
+  dynamic.macos-mac2metal-arm64.instance-type: {{ (index $config "instance-type") | default "mac2metal" | quote }}
+  dynamic.macos-mac2metal-arm64.instance-tag: {{ (index $config "instance-tag") | default (printf "%s-arm64-mac2metal" $environment) | quote }}
+  dynamic.macos-mac2metal-arm64.key-name: {{ default (index $amd "key-name") ((index $config "key-name")) | quote }}
+  dynamic.macos-mac2metal-arm64.aws-secret: {{ (index $config "aws-secret") | default "aws-account" | quote }}
+  dynamic.macos-mac2metal-arm64.ssh-secret: {{ (index $config "ssh-secret") | default "aws-ssh-key" | quote }}
+  dynamic.macos-mac2metal-arm64.security-group-id: {{ default (index $amd "security-group-id") ((index $config "security-group-id")) | quote }}
+  dynamic.macos-mac2metal-arm64.max-instances: {{ (index $config "max-instances") | default "5" | quote }}
+  dynamic.macos-mac2metal-arm64.subnet-id: {{ default (index $amd "subnet-id") ((index $config "subnet-id")) | quote }}
+  dynamic.macos-mac2metal-arm64.disk: {{ index $config "disk" | default "100" | quote }}
+  dynamic.macos-mac2metal-arm64.allocation-timeout: "1200"
+  dynamic.macos-mac2metal-arm64.tenancy: "host"
+  dynamic.macos-mac2metal-arm64.host-resource-group: {{ index $config "host-resource-group" | quote }}
+  dynamic.macos-mac2metal-arm64.license-configuration: {{ index $config "license-configuration" | quote }}
+  {{ end }}
```

we are missing a few options, including the important one of user-data
```suggestion
  {{- if (index $config "iops") }}
  dynamic.macos-mac2metal-arm64.iops: {{ index $config "iops" | quote }}
  {{ end }}
  {{- if (index $config "throughput") }}
  dynamic.macos-mac2metal-arm64.throughput: {{ index $config "throughput" | quote }}
  {{ end }}
  {{- if (index $config "user-data") }}
  dynamic.macos-mac2metal-arm64.user-data: |
  {{- $lines := splitList "\n" (index $config "user-data") }}
  {{- range $line := $lines }}
    {{ $line }}
  {{- end }}
  {{- end }}
  {{ end }}
```